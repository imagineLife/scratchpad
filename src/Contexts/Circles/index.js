import React from 'react';
const CircleContext = React.createContext(); 
const {Provider, Consumer} = CircleContext;
import { 
	getWordsByCount, 
	getLongestThirty, 
	convertStrToWordArr, 
	ingWords,
	arrayOfWords,
	getWordsByLength 
} from '../../lib/stats'

const CirclesProvider = (props) => {
 
	return(<Provider value={{}}>
		{props.children}
	</Provider>)
}

// export default Provider;
export {
	CirclesProvider, 
	Consumer as CirclesConsumer, 
	CircleContext
}