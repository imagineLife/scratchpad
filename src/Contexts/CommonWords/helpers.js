const sortByWordAlpha = (a, b) => {
  if (typeof a === 'string') {
    if (a < b) { return -1; }
    if (a > b) { return 1; }
    return 0;
  }

  if (a.word < b.word) { return -1; }
  if (a.word > b.word) { return 1; }
  return 0;
};

  export { sortByWordAlpha };