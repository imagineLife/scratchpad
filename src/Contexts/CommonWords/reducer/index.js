const reducer = (state, {type, payload}) => {
  switch (type) {
  case 'LONGEST_WORDS':
    return {
      ...state,
      'Longest Words': payload,
    };
    break;

  case 'COMMON_WORDS':
    return {
      ...state,
      'Common Words': payload,
    };
    break;

  case 'ACTION_WORDS':
    return {
      ...state,
      'Action Words': payload,
    };
    break;

  default:
    throw new Error(`CALLED COMMON-WORDS REDUCER WITH BAD TYPE: ${type}`);
    break;
  }
};
module.exports = reducer;