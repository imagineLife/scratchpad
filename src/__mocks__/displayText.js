const txt = 'Chief Justice Roberts, President Carter, President Clinton, President Bush, President Obama, fellow Americans, and people of the world, thank you.';
export default txt;
