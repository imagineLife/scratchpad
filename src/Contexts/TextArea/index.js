import React, {
  useReducer, useState, useEffect, createContext,
} from 'react';
import * as arr from 'd3-array';
import { getQueriedWord } from '../../lib/getQueriedWord';
import { getSentences } from '../../lib/stats';
import reducer from './reducer'
const TextAreaContext = createContext();
const { Provider, Consumer } = TextAreaContext;

const TextAreaProvider = (props) => {
  const initialContext = { text: '', curColor: null };
  const [themeColors] = useState([
    'rgb(166,206,227)',
    'rgb(31,120,180)',
    'rgb(178,223,138)',
    'rgb(51,160,44)',
    'rgb(251,154,153)',
    'rgb(227,26,28)',
    'rgb(253,191,111)',
    'rgb(255,127,0)',
    'rgb(202,178,214)',
    'rgb(106,61,154)',
    'rgb(255,255,153)',
    'rgb(177,89,40)',
  ]);

  const [textAnalyticsStore, textAreaDispatch] = useReducer(reducer, initialContext);
  const [areaData, setAreaData] = useState(null);
  const [themesData, setThemesData] = useState(null);

  /*
		load the text from textFile 'on load'
	*/
  useEffect(() => {
    const textFileURL = process.env.NODE_ENV == 'development' ? '../../data/fullText.txt' : './data/fullText.txt';
    const themesURL = process.env.NODE_ENV == 'development' ? '../../data/themesArrSlim.json' : './data/themesArrSlim.json';

    // console.log('LOADING TEXT in textAreaProvider');
    fetch(textFileURL)
      .then((textFileRes) => textFileRes.text()
      	.then((textRes) => {
	        const sentences = getSentences(textRes);
	        const maxWordCount = arr.max(sentences, (d) => d.wordCount);

	        // update Provider state, triggering reducer with dispatched actions
	        textAreaDispatch({ type: 'TEXT', payload: textRes });
	        textAreaDispatch({ type: 'SENTENCES', payload: sentences });
	        textAreaDispatch({ type: 'MAX_WORDS', payload: maxWordCount });

	        fetch(themesURL)
	          .then((themeRes) => themeRes.json().then(setThemesData));
	      }));
  }, []);

  return (
    <Provider value={{
      textAreaDispatch,
      areaData,
      curColor: textAnalyticsStore.curColor,
      setAreaData,
      themesData,
      ...textAnalyticsStore,
    }}
    >
      {props.children}
    </Provider>
  );
};

// export default Provider;
export {
  TextAreaProvider,
  Consumer as TextAreaConsumer,
  TextAreaContext,
};
