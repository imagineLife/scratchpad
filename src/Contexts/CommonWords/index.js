import React from 'react';
const WordListContext = React.createContext(); 
const {Provider, Consumer} = WordListContext;
import { 
	getWordsByCount, 
	getLongestThirty, 
	convertStrToWordArr, 
	ingWords,
	arrayOfWords,
	getWordsByLength 
} from '../../lib/stats'
import { TextAreaContext } from '../TextArea/'

const CommonWordsProvider = (props) => {
 
 	const initialState = {
 		"Common Words": [],
 		"Longest Words": [],
 		"Action Words": []
 	};

 	const reducer = (state, action) => {
		let resText;
		switch(action.type){
			case "LONGEST_WORDS":
				return {
					...state,
					"Longest Words" : action.payload
				}
				break;
			
			case "COMMON_WORDS":
				return {
					...state,
					"Common Words" : action.payload
				}
				break;

			case "ACTION_WORDS":
				return {
					...state,
					"Action Words" : action.payload
				}
				break;
			
			default:
				return {...state}	
				break;
		}
	}


	let [wordLists, setWordLists] = React.useReducer(reducer, initialState)
 
	let txtVals = React.useContext(TextAreaContext)	

	let [selectedWord, setSelectedWord] = React.useState(null);
	let [commonWords, setCommonWords] = React.useState([])
	let [longestNine, setLongestNine] = React.useState([])
	let [wordListFocus,selectWordList] = React.useState(null)
	const makeCommonWords = (sentencesString)=> {
		let arrayOfWords = convertStrToWordArr(sentencesString)		
		setCommonWords(getWordsByCount(arrayOfWords).slice(0,10))
		setLongestNine(getLongestThirty(arrayOfWords).slice(0,10))
		setWordLists({type: "COMMON_WORDS", payload: getWordsByCount(arrayOfWords).slice(0,10)})
		setWordLists({type: "LONGEST_WORDS", payload: getLongestThirty(arrayOfWords).slice(0,10)})
		setWordLists({type: "WORDS_BY_LENGTH", payload: getWordsByLength(arrayOfWords)})
		setWordLists({type: "ACTION_WORDS", payload: (function getINGWords(){
			let words = ingWords(sentencesString)
			if(!words) {
				return []
			}
			return words.filter((itm,idx) => words.indexOf(itm) === idx)
		})()})
	}

	return(<Provider value={{ 
		selectedWord, 
		setSelectedWord, 
		makeCommonWords, 
		commonWords, 
		longestNine, 
		wordLists,
		selectWordList,
		wordListFocus
	}}>
		{props.children}
	</Provider>)
}

// export default Provider;
export {
	CommonWordsProvider, 
	Consumer as WordListConsumer, 
	WordListContext
}