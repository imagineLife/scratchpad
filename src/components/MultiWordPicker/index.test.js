import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import CommonWords from '.';

// State
import { WordListContext } from '../../Contexts/CommonWords';

// Mock Data
import mockWordLists from '../../__mocks__/wordLists.json';
import displayText from '../../__mocks__/displayText.js';

const mockProps = {
  // wordLists: mockWordLists,
  wordListNames: ['Common Words', 'Longest Words', 'Action Words'],
  selectedWord: null,
  wordListFocus: 'Common Words',
};

describe('<MultiWordPicker />', () => {
  const mockFn1 = jest.fn();
  const mockFn2 = jest.fn();
  const mockFn3 = jest.fn();
  const mockFn4 = jest.fn();

  const theseProps = {
    ...mockProps,
    sortByWordAlpha: mockFn3,
    setSelectedWord: mockFn2,
    makeCommonWords: mockFn1,
    selectWordList: mockFn4,
  };

  const wrapper = mount(
    <WordListContext.Provider
      value={{ ...theseProps }}
    >
      <CommonWords />
    </WordListContext.Provider>,
  );
  it('matches snap', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('handles no wordLists', () => {
  	const pTag = wrapper.find('p').text();
  	expect(pTag).toBe('Missing Word Lists');
  });

  describe('handles props', () => {
  	const thisWrapper = mount(
    <WordListContext.Provider
        value={{ ...theseProps, wordLists: mockWordLists }}
      >
        <CommonWords />
      </WordListContext.Provider>,
    );

    it('builds word-lists from "wordListNames" prop', () => {
    	const wlChoices = thisWrapper.find('#word-lists');
    	const listItems = wlChoices.find('ListItem');
    	expect(listItems).toHaveLength(3);
    });
  });
});
