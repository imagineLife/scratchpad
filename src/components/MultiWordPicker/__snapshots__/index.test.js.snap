// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MultiWordPicker /> matches snap 1`] = `
<CommonWords>
  <p>
    Missing Word Lists
  </p>
</CommonWords>
`;
